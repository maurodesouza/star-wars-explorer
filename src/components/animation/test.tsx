import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Animation, AnimationProps } from '.';

jest.mock('react-lottie', () => ({
  __esModule: true,
  default: function Mock() {
    return <svg data-testid="lottie" />;
  },
}));

const dataMock: AnimationProps = {
  animation: 'loading',
  label: 'foo',
};

describe('<Animation />', () => {
  it('should render the animation styles correct', () => {
    const { container } = renderWithProviders(<Animation {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the animation component correct', () => {
    renderWithProviders(<Animation {...dataMock} />);

    const label = screen.getByText(dataMock.label!);
    const lottie = screen.getByTestId('lottie');

    expect(label).toBeInTheDocument();
    expect(lottie).toBeInTheDocument();
  });
});
