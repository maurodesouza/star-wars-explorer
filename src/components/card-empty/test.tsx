import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { EmptyCard } from '.';

const dataMock = {
  id: 'test',
};

describe('<EmptyCard />', () => {
  it('should render the empty card styles correct', () => {
    const { container } = renderWithProviders(<EmptyCard {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the empty card component correct', () => {
    renderWithProviders(<EmptyCard {...dataMock} />);

    const label = screen.getByText(`${dataMock.id}#`);

    expect(label).toBeInTheDocument();
  });
});
