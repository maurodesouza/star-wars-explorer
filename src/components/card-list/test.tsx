import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';
import { createArray } from 'utils';

import { CardList } from '.';

jest.mock('components', () => ({
  __esModule: true,
  Card: function Mock() {
    return <div data-testid="card" />;
  },
}));

const itemMock = {
  id: '1',
  title: 'C-3PO',
  image: 'image',
  entity: Entities.CHARACTERS,
};

describe('<CardList />', () => {
  describe('styles', () => {
    beforeAll(() => {
      jest.resetAllMocks();
    });

    it('should render the card list styles correct', () => {
      const { container } = renderWithProviders(<CardList />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render the items card correct', () => {
    const counts = [4, 8];

    counts.forEach(count => {
      const items = createArray(count).map((_, index) => ({
        ...itemMock,
        id: String(index),
      }));

      const render = renderWithProviders(<CardList items={items} />);
      const findedItems = render.getAllByTestId(/card/i);

      expect(findedItems.length).toBe(count);

      cleanup();
    });
  });

  it('should render the fills correct', () => {
    const minCountToFill = 4;
    const counts = [1, 3, 4, 6];

    counts.forEach(count => {
      const result = count - minCountToFill;
      const lackForFill = result <= 0 ? Math.abs(result) : 0;

      const items = createArray(count).map((_, index) => ({
        ...itemMock,
        id: String(index),
      }));

      const { container } = renderWithProviders(<CardList items={items} />);
      const findedItems = container.firstElementChild!.querySelectorAll(
        'div:not([data-testId]'
      );

      expect(findedItems.length).toBe(lackForFill);
      cleanup();
    });
  });
});
