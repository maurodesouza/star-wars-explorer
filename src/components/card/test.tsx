/* eslint-disable @typescript-eslint/no-var-requires */

import { act, fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';
import { Card } from '.';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const push = jest.fn();

useRouter.mockImplementation(() => ({
  push,
}));

const mockData = {
  id: '1',
  label: 'C-3PO',
  image:
    'https://raw.githubusercontent.com/tbone849/star-wars-guide/master/build/assets/img/characters/2.jpg',
};

describe('<Card />', () => {
  it('should render the card correct', () => {
    renderWithProviders(<Card {...mockData} />);

    expect(
      screen.getByRole('img', { name: mockData.label })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: mockData.label })
    ).toBeInTheDocument();
  });

  it("should render the no image found element when can't load image url", async () => {
    renderWithProviders(<Card {...mockData} />);

    const image = screen.getByRole('img', { name: mockData.label });

    act(() => {
      fireEvent.error(image);
    });

    const noImageFoundElement = screen.getByText(/no found/i);

    expect(noImageFoundElement).toBeInTheDocument();
  });

  it('should render the card styles correct', () => {
    const { container } = renderWithProviders(<Card {...mockData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should navigate to entity page', () => {
    const { container } = renderWithProviders(<Card {...mockData} />);

    fireEvent.click(container.firstChild!);

    expect(push).toBeCalledTimes(1);
    expect(push).toBeCalledWith(`/${Entities.CHARACTERS}/${mockData.label}`);
  });
});
