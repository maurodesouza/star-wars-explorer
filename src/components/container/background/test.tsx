import { renderWithProviders } from 'utils/test/helpers';
import { Background } from '.';

describe('components/container', () => {
  describe('<Background />', () => {
    it('should render the background styles correct', () => {
      const { container } = renderWithProviders(<Background />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
