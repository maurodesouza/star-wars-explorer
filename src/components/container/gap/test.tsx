import { renderWithProviders } from 'utils/test/helpers';
import { Gap } from '.';

describe('components/container', () => {
  describe('<Gap />', () => {
    it('should render the gap styles correct', () => {
      const { container } = renderWithProviders(<Gap />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
