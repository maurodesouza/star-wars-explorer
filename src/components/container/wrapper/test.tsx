import { renderWithProviders } from 'utils/test/helpers';
import { Wrapper } from '.';

describe('components/container', () => {
  describe('<Wrapper />', () => {
    it('should render the wrapper styles correct', () => {
      const { container } = renderWithProviders(<Wrapper />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
