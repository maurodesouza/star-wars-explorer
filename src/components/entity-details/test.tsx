import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';

import { mock } from './mock';
import { EntityDetails, EntityDetailsProps } from '.';

const dataMock: EntityDetailsProps = {
  ...mock,
  id: '1',
  title: 'label',
  entity: Entities.CHARACTERS,
};

describe('<EntityDetails />', () => {
  it('should render the entity details styles correct', () => {
    const { container } = renderWithProviders(<EntityDetails {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the entity details component correct', () => {
    renderWithProviders(<EntityDetails {...dataMock} />);

    const image = screen.getByRole('img', { name: dataMock.title });

    expect(image).toBeInTheDocument();

    Object.entries(mock.extras)
      .reduce((arr, [key, value]) => [...arr, `${key}:`, value], [] as string[])
      .forEach(text => {
        const el = screen.getByText(text);

        expect(el).toBeInTheDocument();
      });
  });

  it('should no render content if no extras was passed', () => {
    const { container } = renderWithProviders(
      <EntityDetails {...dataMock} extras={undefined} />
    );

    const component = container.children[0];
    const content = component.children[1];

    expect(content.children.length).toBe(0);
  });
});
