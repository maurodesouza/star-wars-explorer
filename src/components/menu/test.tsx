import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Menu } from '.';

describe('<Menu />', () => {
  it('should render the menu correct', () => {
    renderWithProviders(<Menu />);

    expect(screen.getByRole('img', { name: /star wars/i })).toBeInTheDocument();

    expect(screen.getByLabelText(/open menu/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/close menu/i)).toBeInTheDocument();

    const exploreLinks = screen.getAllByText(/explore/i);

    expect(exploreLinks).toHaveLength(2);

    exploreLinks.forEach(link => {
      expect(link).toHaveAttribute('href', '/');
    });
  });

  it('should open/close full menu', () => {
    renderWithProviders(<Menu />);

    const openMenuElement = screen.getByLabelText(/open menu/i);
    const closeMenuElement = screen.getByLabelText(/close menu/i);

    const fullMenuElement = closeMenuElement.parentElement!.parentElement!;

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true');
    expect(fullMenuElement).toHaveStyle({ opacity: 0 });

    fireEvent.click(openMenuElement);

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('false');
    expect(fullMenuElement).toHaveStyle({ opacity: 1 });

    fireEvent.click(closeMenuElement);

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true');
    expect(fullMenuElement).toHaveStyle({ opacity: 0 });
  });

  it('should render the menu styles correct', () => {
    const { container } = renderWithProviders(<Menu />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
