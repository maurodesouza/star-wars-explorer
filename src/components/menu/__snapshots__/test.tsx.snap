// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render the menu styles correct 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  width: min(100%,8rem);
}

.c11 {
  text-align: center;
  width: min(100%,57rem);
  color: #FFF;
  font-size: 1.2rem;
}

.c12 {
  margin-top: calc(1.2rem / 2);
}

.c12 a {
  color: #FFF;
}

.c12 a:hover {
  color: #FFE81F;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  color: #FFF;
  font-weight: 700;
  font-size: 4.0rem;
}

.c4 + .c4 {
  margin-top: 2.4rem;
}

.c7 {
  color: #FFF;
  width: 2.4rem;
  height: 2.4rem;
  cursor: pointer;
}

.c9 {
  position: fixed;
  inset: 0;
  height: 100vh;
  width: 100%;
  background-color: #000;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  opacity: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  padding: 0 2.4rem 2.4rem;
  z-index: 20;
}

.c9 .c6 {
  position: absolute;
  top: 2.4rem;
  right: 2.4rem;
}

.c9 .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: translateY(3.2rem);
  -ms-transform: translateY(3.2rem);
  transform: translateY(3.2rem);
}

.c9 .c10 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: translateY(3.2rem);
  -ms-transform: translateY(3.2rem);
  transform: translateY(3.2rem);
}

@media (min-width:768px) {
  .c1 {
    width: min(100%,12.5rem);
  }
}

@media (min-width:768px) {
  .c11 {
    font-size: 1.6rem;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: initial;
    -webkit-justify-content: initial;
    -ms-flex-pack: initial;
    justify-content: initial;
  }
}

@media (min-width:768px) {
  .c5 {
    position: relative;
    font-size: 2.4rem;
  }

  .c4 + .c4 {
    margin-top: 0;
    margin-left: calc(4.0rem);
  }

  .c5::after {
    content: '';
    position: absolute;
    display: block;
    background-color: #FFE81F;
    width: 100%;
    bottom: -8px;
    height: 2px;
    -webkit-transform: scaleX(0);
    -ms-transform: scaleX(0);
    transform: scaleX(0);
    -webkit-transition: -webkit-transform 0.3s;
    -webkit-transition: transform 0.3s;
    transition: transform 0.3s;
  }

  .c5:hover::after {
    -webkit-transform: scaleX(1);
    -ms-transform: scaleX(1);
    transform: scaleX(1);
  }
}

@media (min-width:768px) {
  .c7 {
    display: none;
  }
}

@media (min-width:768px) {
  .c9 {
    display: none;
  }
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-label="Star wars"
      fill="none"
      height="50"
      role="img"
      viewBox="0 0 106 50"
      width="106"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0_6_76)"
      >
        <path
          d="M19.813 27.1386L21.2588 31.8294C22.0411 34.3661 22.7395 36.3943 22.8305 36.3943C22.832 36.3942 22.8334 36.3935 22.8345 36.3924C22.9346 36.2948 25.7645 27.2155 25.7645 27.2155H31.2493L24.6322 48.5162H20.82C20.82 48.5162 16.751 35.533 16.7677 35.4838L12.5203 48.3528H8.74649L2.21276 27.1386L7.67087 27.1425C7.67087 27.1425 10.6607 36.5052 10.7006 36.5052L10.7009 36.5048L13.6843 27.1383L19.813 27.1386ZM0.349366 25.6633L0.949875 27.6132L7.4836 48.8273L7.79127 49.8266H13.454L13.7739 48.8575L16.723 39.9223C16.9772 40.7392 17.2518 41.6196 17.5265 42.4992C18.2031 44.6659 18.8808 46.8322 19.5595 48.9981L19.8703 49.99H25.5849L25.8941 48.9948L32.5112 27.6943L33.1177 25.7416H24.813L24.5031 26.7357C23.9346 28.5596 23.3622 30.382 22.7862 32.203C22.6971 31.9202 22.6089 31.6371 22.5215 31.3537L21.0758 26.6631L20.768 25.6644H12.7396L12.4259 26.6493L10.6828 32.1221C10.0962 30.2995 9.51166 28.4762 8.92909 26.6521L8.61524 25.6694L7.67187 25.6686L2.21376 25.6648L0.349366 25.6633ZM104.284 27.0561L104.272 32.6392C104.272 32.6392 98.8283 32.6193 96.0609 32.6193C95.1402 32.6193 94.5154 32.6219 94.4859 32.6276C94.0094 32.7236 93.7054 33.8441 93.8407 34.3357C93.907 34.5846 94.4065 35.3968 94.9465 36.1432C95.4849 36.8897 96.3708 38.1111 96.9107 38.8567C98.2979 40.7692 98.4881 41.1467 98.6167 42.2449C98.8252 44.0288 98.0617 45.959 96.6442 47.2321C95.2233 48.5088 95.2858 48.5178 84.7403 48.5178H84.2772C77.8634 48.5158 74.4715 48.4572 73.9777 48.3358C73.1005 48.1166 71.7025 46.8564 68.0091 42.9415C66.6805 41.5347 65.4558 40.3029 65.3929 40.3026C65.3913 40.3026 65.3906 40.3038 65.3906 40.3059L65.3469 48.5186L59.3605 48.4895L59.3235 27.3065H72.6748C76.6986 27.4669 79.4218 31.8957 78.8798 34.72C78.7571 35.3609 78.4434 36.2898 78.1827 36.7834C77.6473 37.7966 76.2848 39.0973 75.179 39.6557C74.7706 39.8597 74.4371 40.0973 74.4371 40.1821C74.4371 40.4971 75.7814 41.9595 76.2848 42.1919C76.7151 42.3906 77.9674 42.4349 83.4908 42.4349L85.2596 42.4352C90.8116 42.4352 90.5321 42.391 90.961 41.3775C91.2614 40.6721 91.0331 40.2502 88.9202 37.6211C86.2435 34.6787 86.5237 33.2693 86.5055 32.2284C86.4673 30.981 87.3563 27.0574 91.9773 27.0574L104.284 27.0561ZM68.419 36.8423C70.9821 36.8423 71.5026 36.7985 72.0479 36.5367C73.8891 35.6522 73.9875 33.0401 72.2215 31.8809C71.8346 31.6265 71.3375 31.5847 68.5592 31.5679L65.3443 31.5491C65.3709 31.5766 65.3341 36.6284 65.3341 36.6284C65.3341 36.6284 66.1394 36.8423 68.419 36.8423ZM105.622 25.5822H91.9773C89.3036 25.5822 87.7222 26.7479 86.8669 27.7261C85.4744 29.3166 85.1472 31.3194 85.1722 32.2636C85.1736 32.3423 85.1729 32.423 85.1719 32.5072C85.1599 34.0416 85.364 35.7787 87.9501 38.6321C88.4608 39.2679 89.3158 40.3341 89.6396 40.8639C88.9329 40.9599 87.1415 40.9599 85.2594 40.9599H84.4605L83.4909 40.9595C80.1717 40.9595 77.3726 40.9488 76.7993 40.8239C76.7189 40.7665 76.6065 40.6666 76.4828 40.548C77.6296 39.7681 78.7891 38.5584 79.3351 37.5245C79.6662 36.8957 80.0318 35.8217 80.1852 35.0251C80.5256 33.2518 79.9644 31.1127 78.6844 29.3013C77.1907 27.188 75.0179 25.9236 72.7233 25.8322L72.6991 25.8311H57.9869L57.9897 27.3083L58.0267 48.4911L58.0294 49.9556L59.355 49.962L65.3414 49.9911L66.6732 49.9974L66.6813 48.5259L66.707 43.606C66.8291 43.7341 66.9555 43.8674 67.0856 44.0047C71.126 48.2862 72.4815 49.4727 73.683 49.7728C74.0132 49.8539 74.5592 49.9887 84.2762 49.9917H84.7393C90.0005 49.9917 92.641 49.9917 94.2115 49.8242C96.0107 49.6321 96.5896 49.1812 97.4569 48.4007L97.4859 48.3747C99.2676 46.7741 100.208 44.3529 99.9398 42.0555C99.7649 40.5614 99.407 39.9407 97.9517 37.9346L97.5191 37.3377C97.0085 36.6334 96.4984 35.9287 95.9886 35.2237C95.6577 34.766 95.397 34.3598 95.2408 34.0936C95.4466 34.093 95.7144 34.0927 96.0609 34.0927C98.795 34.0927 104.213 34.1124 104.268 34.1127L105.603 34.1173L105.607 32.642L105.619 27.0585L105.622 25.5822ZM66.6767 35.3147L66.6828 34.1065C66.6845 33.738 66.6862 33.3688 66.6863 33.0306L68.5521 33.0414C69.6244 33.0477 71.2376 33.0576 71.5521 33.1562C71.9533 33.4254 72.1775 33.8417 72.1531 34.2728C72.1398 34.5057 72.0394 34.9339 71.5158 35.1848C71.2239 35.325 70.7588 35.3685 68.4193 35.3685C67.6769 35.3685 67.1044 35.3447 66.6767 35.3147ZM47.1936 26.9691L53.9126 48.3253L48.4376 48.3528L47.5288 45.1453L37.7398 45.0585L36.876 48.2698L31.4563 48.207L38.09 26.9782L47.1936 26.9691ZM42.7909 30.1002C42.7771 30.1615 41.9331 32.7199 41.0863 35.2778C40.2324 37.8571 39.3758 40.4363 39.3639 40.4562C39.3866 40.4488 42.4183 40.4415 44.3615 40.4415C45.2708 40.4415 45.9415 40.4433 45.954 40.4466L45.9542 40.4475V40.447L45.954 40.4466C45.92 40.3497 42.8007 30.2249 42.7909 30.1002L42.7911 30.0991L42.7909 30.1002ZM48.1419 25.4942L47.1926 25.4953L38.089 25.5047L37.1392 25.5058L36.8292 26.4981L30.1954 47.7269L29.5912 49.6601L31.4426 49.6813L36.8623 49.7441L37.8684 49.7557L38.1551 48.6905L38.7331 46.5419L46.5465 46.611L47.1649 48.794L47.4592 49.8329L48.4443 49.8281L53.9191 49.8006L55.7858 49.7909L55.1726 47.843L48.4536 26.4868L48.1419 25.4942ZM41.2842 38.9696C41.5368 38.2088 41.8772 37.1807 42.3393 35.7845L42.7384 34.5787L43.1001 35.757C43.2078 36.1082 43.6812 37.6493 44.0873 38.968C42.8799 38.968 41.9711 38.9682 41.2842 38.9696ZM49.4582 1.479V6.74803H39.9061V22.863H34.2112V6.73532L23.0615 6.77014C21.7174 6.77014 21.5373 7.95291 21.5373 8.40612C21.5373 8.96268 21.8658 9.51888 23.6702 12.0429C24.8408 13.6823 25.9164 15.3185 26.0582 15.6759C26.832 17.6437 25.8314 20.7091 24.0087 21.9545C22.7096 22.8412 22.9316 22.8632 13.2846 22.8632H1.33409V16.9436H17.6601L18.142 16.5733C18.4672 16.3229 18.6556 15.9784 18.7273 15.5066C18.8274 14.8286 18.7757 14.7402 16.7596 12.2311C14.3765 9.26482 14.1247 8.72336 14.2481 6.82964C14.3882 4.6686 16.1325 1.52911 19.1809 1.52911L49.4582 1.479ZM50.7923 0.00292969L49.456 0.0053247L19.1788 0.0550673C17.4348 0.0550673 15.815 0.887796 14.6192 2.39905C13.6496 3.62438 13.0133 5.24175 12.9172 6.72445C12.7549 9.2136 13.2736 10.1132 15.7615 13.2098C16.4836 14.1083 17.1701 14.9629 17.3993 15.3257C17.3968 15.3357 17.3938 15.3454 17.3902 15.355C17.3877 15.3574 17.3839 15.3603 17.3786 15.3646L17.2415 15.4698H0V24.337H13.2846C18.041 24.337 20.3886 24.3322 21.7717 24.2198C23.3543 24.092 23.8593 23.7921 24.6157 23.2729L24.7101 23.208C27.0884 21.5831 28.3141 17.713 27.2819 15.0884C26.9634 14.2855 24.9467 11.4505 24.7178 11.13C24.0847 10.2442 23.0398 8.78287 22.8744 8.35987C22.8784 8.3245 22.8862 8.28747 22.8946 8.26205C22.9221 8.25431 22.9748 8.24381 23.0615 8.24381L32.8771 8.21304V24.3367H41.2402V8.2217H50.7923V0.00292969V0.00292969ZM64.977 1.65015L71.6775 23.0616L66.2127 23.0284L65.3256 19.8209L55.5802 19.773L54.7095 22.8628H49.1448L55.7718 1.65015H64.977ZM57.1708 15.2135L63.8082 15.2043V15.2045C63.8311 15.2045 60.5827 4.66621 60.561 4.65884C60.5373 4.65792 57.096 15.2135 57.1708 15.2135ZM65.9276 0.176292H54.8214L54.5114 1.16949L47.8841 22.3823L47.2734 24.3368H55.6924L55.9842 23.3009L56.5614 21.2521L64.338 21.2902L64.9382 23.4592L65.2253 24.4962L66.2071 24.5023L71.672 24.5356L73.5559 24.547L72.9404 22.5802L66.2399 1.16838L65.9276 0.176292ZM59.0241 13.7372C59.5206 12.181 60.0217 10.6266 60.5273 9.07396C61.0066 10.6263 61.484 12.1794 61.9595 13.7332L59.0241 13.7372ZM83.7958 1.65015C88.1186 1.65015 91.1183 1.71831 91.6954 1.82885C93.0295 2.0853 94.2351 2.8716 95.1387 4.08053C96.0843 5.344 96.3012 5.94865 96.4246 7.68024C96.6016 10.1252 95.5491 12.2346 93.4728 13.6031C92.8676 14.0029 92.2459 14.3916 92.0905 14.469C91.852 14.5886 91.884 14.7032 92.3106 15.2559C92.5854 15.6116 93.0627 16.1182 93.3696 16.3813L93.9297 16.8622L104.282 17.0261L104.384 22.8628H98.0365C93.1274 22.8628 91.5567 22.8129 91.0958 22.6419C90.4072 22.384 88.7501 20.8327 85.3808 17.2918L82.9761 14.764L83.0015 22.8628H76.8238V1.65015H83.7958ZM82.8857 11.1878H86.1659C89.3579 11.1878 89.4578 11.1766 89.9133 10.7993C90.7272 10.125 91.0154 9.43026 90.9288 8.36356C90.841 7.26351 90.4569 6.69092 89.4947 6.23034C88.9527 5.96854 88.4306 5.92635 85.8724 5.92635H82.8857V11.1878ZM83.7958 0.176292H75.4912V24.3367H84.3412L84.3367 22.8578L84.3217 18.212L84.4622 18.3594C88.6285 22.7388 89.8577 23.734 90.6685 24.0375C91.2909 24.2696 92.3331 24.3367 98.0374 24.3367H105.744L105.718 22.8346L105.617 16.998L105.593 15.5733L104.303 15.5528L94.3968 15.3961L94.1894 15.2177C94.1233 15.1606 94.0431 15.0854 93.9577 15.0003L94.159 14.8682C96.6608 13.2188 97.972 10.5559 97.7557 7.56289C97.6118 5.54389 97.2855 4.63434 96.168 3.14132C95.07 1.67318 93.563 0.691404 91.9244 0.37692C91.0134 0.202453 86.4408 0.176292 83.7958 0.176292ZM84.2198 7.40003H85.8724C88.2113 7.40003 88.6749 7.44295 88.9605 7.58076C89.5025 7.83997 89.5576 7.96599 89.5999 8.49381C89.6426 9.01869 89.5839 9.22963 89.1126 9.61965L89.0754 9.65023C88.7644 9.71398 87.6104 9.71398 86.1659 9.71398H84.2198V7.40003Z"
          fill="#FFE81F"
        />
      </g>
      <defs>
        <clippath
          id="clip0_6_76"
        >
          <rect
            fill="white"
            height="50"
            width="105.744"
          />
        </clippath>
      </defs>
    </svg>
  </div>
  <nav
    class="c2 c3"
  >
    <a
      class="c4 c5"
      href="/"
    >
      Explore
    </a>
    <a
      class="c4 c5"
      href="/achievements"
    >
      Achievements
    </a>
    <a
      class="c4 c5"
      href="/favorites"
    >
      Favorites
    </a>
  </nav>
  <div
    class="c6 c7"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c8"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    aria-hidden="true"
    class="c9"
  >
    <div
      class="c6 c7"
    >
      <svg
        aria-hidden="true"
        aria-label="Close menu"
        class="c8"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <nav
      class="c2 c3"
    >
      <a
        class="c4 c5"
        href="/"
      >
        Explore
      </a>
      <a
        class="c4 c5"
        href="/achievements"
      >
        Achievements
      </a>
      <a
        class="c4 c5"
        href="/favorites"
      >
        Favorites
      </a>
    </nav>
    <footer
      class="c10 c11"
    >
      <strong
        class=""
      >
        Data provided by SWAPI
      </strong>
      <p
        class="c12"
      >
        Made with ❤ by
         
        <a
          href="https://www.linkedin.com/in/maurodesouzaa/"
          rel="noreferrer"
          target="_blank"
        >
          Mauro de Souza
        </a>
         
      </p>
    </footer>
  </div>
</header>
`;
